### PR TITLE
Ftp combine move remove

### PR DESCRIPTION
**Pull request by wrp801**
Adding the ability to move and remove files on an FTP server

### CODE REVIEW

File: ftp_blueprints/move_file.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+                destination_file_name = args.destination_file_name, 
+                source_full_path = key_name,
+                file_number = None if len(matching_file_names) == 1 else index
+
+            )
+            if len(destination_full_path.split('/')) > 1:
+                path, file_name = destination_full_path.rsplit('/', 1)
+                ftp_create_new_folders(client=client, destination_path=path)
+            file_name = destination_full_path.rsplit('/', 1)[-1]
+            print(f'Moving file {index+1} of {len(matching_file_names)}')
```
Comment (blakeburch) [inline ftp_blueprints/move_file.py]: No need for +1 here since you started at 1. It ends up saying "moving file 2 of 1".

**Review by blakeburch — CHANGES_REQUESTED** [review]
Code _looks_ fine. Regex functionality has some issues.

1. When moving a file using regex, the matching logic appears to be wrong. If I leave the source_folder_name blank, it only searches for files in the root directory, rather than using that input as a "filter" value. So if a file should match in different subdirectories, it doesn't find it.

This happens for both move and delete functions.

**Comment by wrp801** [comment]
@blakeburch just so I understand here, you’re saying that if I provide the pattern as `.csv` and it doesn’t find any matches in the root directory, it should also search the directories within the root to see if there are matches?

**Comment by blakeburch** [comment]
yes. It should recursively look through all subdirectories to see if there are matches.

This is how every other Blueprint works with regex matching logic.

This situation should be accounted for in our documentation. See “enumerated destination files” example. <https://www.shipyardapp.com/docs/reference/blueprints/blueprint-library/match-type/#regex-match>

**Comment by wrp801** [comment]
Okay cool, that makes sense

**Review by blakeburch — APPROVED** [review]
Good to launch.

**Pull request merged**